sync/app.py

```diff
@@ -0,0 +1,75 @@
+import glob
+import gzip
+import json
+import sys
+from zipfile import ZipFile
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import boto3
+from cryptography.fernet import Fernet
+
+ROOT = Path('/tmp/')
+s3 = boto3.client('s3')
+secret_manager = boto3.client('secretsmanager')
+encryption_secret_key_id = 'arn:aws:secretsmanager:us-east-1:370358067229:secret:efs/problem/encryptionKey-xTnJWC'
+
+
+def sync_handler(event, context):
+    print('Event:', type(event), event)
+    print('Context:', context)
+
+    bucket = event['Records'][0]['s3']['bucket']['name']
+    key = event['Records'][0]['s3']['object']['key']  # some-bucket/some-folder/img.jpg
+    problem = key.split('.')[0]
+    problem_file = f'/mnt/efs/{problem}.gz.fer'
+    zip_path = ROOT / f'{problem}.zip'
+    print('problem_file', problem_file, 'zip:', zip_path)
+
+    s3.download_file(bucket, key, str(zip_path))
+    print('download size:', zip_path.stat().st_size)
+
+    with TemporaryDirectory() as extraction_dir, ZipFile(zip_path, 'r') as zip_ref:
+        zip_ref.extractall(extraction_dir)
+        targets = glob.glob(f'{extraction_dir}/**/*.ans.txt', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.out.txt', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.a', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.ans', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.out', recursive=True)
+        targets = sorted(targets)
+        print('targets:', targets)
+        inputs = [t.replace('.a', '') if t.endswith('a') else t.replace('ans', 'in') for t in targets]
```

Comment:
Logic here should be different. 
Remove the .txt suffix if it exists. 
Then if it ends with .a, .out or .ans remove that suffix and replace it with empty string or '.in'
Probably create a separate function for this. That function can be unit tested as well.

Reply:
didn't quite understand what's the advantage of removing suffixes instead of including all?